src/Farmer/Builders/Builders.Dns.fs

```diff
@@ -183,7 +218,21 @@ type DnsNsRecordBuilder() =
 
     /// Add NSD names (Subdomain NameServers)
     [<CustomOperation "add_nsd_names">]
-    member _.RecordNsdNames(state:NsRecordProperties, nsdNames) = { state with NsdNames = state.NsdNames @ nsdNames }
+    member _.RecordNsdNames(state:NsRecordProperties, nsdNames) =
+        match state.NsdNames with
+        | NsRecords.SourceZone _ ->
+            failwith "Cannot add_nsd_names when using 'reference_nsd_names' to reference another zone's nameservers."
+        | NsRecords.Records existingNsdNames ->
+            { state with NsdNames = NsRecords.Records(existingNsdNames @ nsdNames) }
+
+    /// Reference another DNS zone's nameservers.
+    [<CustomOperation "reference_nsd_names">]
```

Comment:
If you use first reference_nsd_names and then add_nsd_names, it will fail.
I think it should be consistent, so that if you use first add_nsd_names and then reference_nsd_names it should also fail.
So checking the state.NsdNames in these methods like in 8 lines above this.